packages/block-editor/src/components/block-lock/toolbar.js

```diff
@@ -3,65 +3,46 @@
  */
 import { __ } from '@wordpress/i18n';
 import { ToolbarButton, ToolbarGroup } from '@wordpress/components';
-import { focus } from '@wordpress/dom';
 import { useReducer, useRef, useEffect } from '@wordpress/element';
-import { lock } from '@wordpress/icons';
+import { lock, unlock } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
 import BlockLockModal from './modal';
 import useBlockLock from './use-block-lock';
 
-export default function BlockLockToolbar( { clientId, wrapperRef } ) {
-	const { canEdit, canMove, canRemove, canLock } = useBlockLock( clientId );
+export default function BlockLockToolbar( { clientId } ) {
+	const { isLocked } = useBlockLock( clientId );
 
 	const [ isModalOpen, toggleModal ] = useReducer(
 		( isActive ) => ! isActive,
 		false
 	);
 
-	const lockButtonRef = useRef( null );
-	const isFirstRender = useRef( true );
+	const hasLockButtonShown = useRef( false );
 
-	const shouldHideBlockLockUI =
```

Comment:
I think there's a use-case where folks don't want people to be able to unlock and I believe there's a setting for that, so we need to make sure that this setting is still supported.

Reply:
The behavior on trunk is to not show the Lock Icon at all if `canLockBlocks` is false. So, the block is locked, but the user has no indicator. I'll match the behavior on trunk for now, then follow up with another PR that displays the lock icon without it opening a modal.

Reply:
Implemented the idea on showing a disabled lock icon for locked blocks when the user cannot edit it: https://github.com/WordPress/gutenberg/pull/57274

Reply:
@jeryj, that's how I initially implemented the toolbar button but then reverted the change based on suggestion - https://github.com/WordPress/gutenberg/pull/39566#issuecomment-1072778447.